gtsam/discrete/DiscreteConditional.cpp

```diff
@@ -91,7 +91,7 @@ bool DiscreteConditional::equals(const DiscreteFactor& other,
     return false;
   else {
     const DecisionTreeFactor& f(
-        static_cast<const DecisionTreeFactor&>(other));
+        dynamic_cast<const DecisionTreeFactor&>(other));
```

Comment:
What's up with this? Was this needed?
Generally, static_cast is considered safer than dynamic_cast, hence asking.

Reply:
Oh yeah this is a clang-tidy false positive. I can revert this one